### ISSUE TITLE
[Bug]: the inability to launch a casino robbery

### ISSUE THREAD

**Issue by FudayVD1** [Yimura/YimMenu#3335]
### Describe the bug

when activating one of the functions (probably something with protection), there is an endless inscription "starting a session" at the start of a casino robbery

### Steps To Reproduce

idk, this was not the case in previous versions of the menu

### Expected Behavior

endless running session when starting a casino robbery and a robbery in an upper class apartment

### Logs

[cout.txt](https://github.com/user-attachments/files/16162504/cout.txt)

### Screenshots / Video

no video

### Store

Rockstar

### Language

Russian

### YimMenu Language

English (US)

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by pierrelasse** [comment]
This mb has to do with it
![image](https://github.com/YimMenu/YimMenu/assets/76223390/49f7200f-441c-4b3f-a432-9c07885e8b24)

**Comment by OXP3573** [comment]
> This mb has to do with it ![image](https://private-user-images.githubusercontent.com/76223390/347750225-49f7200f-441c-4b3f-a432-9c07885e8b24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2OTYyNzAsIm5iZiI6MTcyMDY5NTk3MCwicGF0aCI6Ii83NjIyMzM5MC8zNDc3NTAyMjUtNDlmNzIwMGYtNDQxYy00YjNmLWE0MzItOWMwNzg4NWU4YjI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzExVDExMDYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlNTY3MTI1ZWEzOTk4NWNlZTdiNjQ1MjQxNzU0OTg1YWJlYTU4YzFjODk4NzBmNWExZWY0NTg1OTllNjliYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yKkG4uoRr-8CuphbFK_ML7buonRbQRt0EvKFi8m-cqI)

No, this was last month when an RCE exploit was identified with the Casino Heist. They issued hotfix code through bg_script to disable access. It has since been fixed and re-enabled.

**Comment by OXP3573** [comment]
> ### Describe the bug
> when activating one of the functions (probably something with protection), there is an endless inscription "starting a session" at the start of a casino robbery
> 
> ### Steps To Reproduce
> idk, this was not the case in previous versions of the menu
> 
> ### Expected Behavior
> endless running session when starting a casino robbery and a robbery in an upper class apartment
> 
> ### Logs
> [cout.txt](https://github.com/user-attachments/files/16162504/cout.txt)
> 
> ### Screenshots / Video
> no video
> 
> ### Store
> Rockstar
> 
> ### Language
> Russian
> 
> ### YimMenu Language
> English (US)
> 
> ### Additional context
> _No response_
> 
> ### Build
> * [x]  I've confirmed the issue exists on the latest version of YimMenu
> 
> ### Coload
> * [x]  I've tested this without co-loading anything.

I have tested this on the latest Nightly Build and the Casino Heist is working correctly. The only issue I was aware of for Jobs was a script event block for races / series.

**Comment by gir489returns** [comment]
@maybegreat48 told me he also encountered this issue.

**Comment by OXP3573** [comment]
Have you tried without these loaded?
billionaire_services.lua
CasinoPacino.lua
CasinoPacinoJson.lua

Considering the names, I think it is worth a try. There could be an issue
but I have done the Casino Heist 3-4 times on various recent builds and 
have not experienced issues with session launch. At worst there has been 
a few instances of things not working correctly, but that is more down to sync
than anything to do with the menu.

I've noticed a few of these in your log:
EXCEPTION_ACCESS_VIOLATION

Your casino mod scripts could be causing issues in the background, especially if you are
using them before / during the mission and they're spaghetti code.

**Comment by OXP3573** [comment]
Example:

//url:bugstar:3280561 - [BG SCRIPT][PUBLIC][REPORTED] "Insurance Fraud" Mod is causing players to be put into Bad Sport lobbies and lose GTA$
PROC FIX_FOR_3280561()
	IF IS_PC_VERSION()
		INT iPlayer
		REPEAT COUNT_OF(GlobalplayerBD_FM) iPlayer
			SET_BIT(GlobalplayerBD_FM[iPlayer].iPersonalVehicleFlagsBS, PERSONAL_VEHICLE_BD_FLAG_IN_VEHICLE_WITH_BOUNTY)
		ENDREPEAT
	ENDIF
ENDPROC

BOOL bDoneEnableDecreasingDupeValueCheck
//url:bugstar:3533670 - [BG SCRIPT ONLY] Decreasing dupe sell values for repeat offenders.
FUNC BOOL IS_REPEAT_OFFENDER_1()
	INT iCurrentPeak = GET_MP_INT_PLAYER_STAT(MPPLY_CHAR_PEAK_EXPLOIT_LEVEL)
	IF (iCurrentPeak <  g_sMPTunables.iRepeatOffenderMinExploitLevel1)
		RETURN(FALSE)
	ENDIF
	INT iPrevious[4]
	INT i	
	iPrevious[0] = GET_MP_INT_PLAYER_STAT(MPPLY_PREVSEASONEXPLOITLEVEL)
	iPrevious[1] = GET_MP_INT_PLAYER_STAT(MPPLY_PREVSEASON2EXPLOITLEVEL)
	iPrevious[2] = GET_MP_INT_PLAYER_STAT(MPPLY_PREVSEASON3EXPLOITLEVEL)
	iPrevious[3] = GET_MP_INT_PLAYER_STAT(MPPLY_PREVSEASON4EXPLOITLEVEL)
	REPEAT (g_sMPTunables.iRepeatOffenderNumSeasons1-1) i
		IF (iPrevious[i] < g_sMPTunables.iRepeatOffenderMinExploitLevel1)
			RETURN(FALSE)
		ENDIF
	ENDREPEAT
	RETURN(TRUE)
ENDFUNC
